bluetooth_mesh/messages/config.py

```diff
@@ -543,25 +543,66 @@ def AddressTypeValidator(subcons, *allowed_types):
     (obj_ & 0xF000) == 0x00
 )
 
-CompositionDataElement = Struct(
+CompositionDataPage0Element = Struct(
     "location" / GATTNamespaceDescriptorAdapter,
     "sig_number" / Rebuild(Int8ul, len_(this["sig_models"])),
     "vendor_number" / Rebuild(Int8ul, len_(this["vendor_models"])),
     "sig_models" / SIGModelId[this["sig_number"]],
     "vendor_models" / VendorModelId[this["vendor_number"]],
 )
 
-CompositionData = Struct(
+CompositionDataPage0 = Struct(
     "cid" / Int16ul,
     "pid" / Int16ul,
     "vid" / Int16ul,
     "crpl" / Int16ul,
     "features" / Int16ul,  # TODO should be parsed
-    "elements" / GreedyRange(CompositionDataElement),
+    "elements" / GreedyRange(CompositionDataPage0Element),
+)
+
+ExtendedModelLongFormat = Struct(
+    "element_offset" / Int8sl,
+    "model_item_index" / Int8ul
+)
+
+ElementOffsetRepresentedValue = {
+    0: 0,
+    1: 1,
+    2: 2,
+    - 4: 4,
+    - 3: 5,
+    - 2: 6,
+    - 1: 7
+}
+
+ExtendedModelShortFormat = BitStruct(
+    "model_item_index" / BitsInteger(5),
+    "element_offset" / Mapping(BitsInteger(3), ElementOffsetRepresentedValue),
+)
+
+ModelRelationItem = BitStruct(
+    "extended_items_count" / Rebuild(BitsInteger(6), len_(this["extended_models_items"])),
+    "format" / Flag,
+    "corresponding_present" / Flag,
+    "corresponding_id" / If(this.corresponding_present, BitsInteger(8)),
+    "extended_models_items" / IfThenElse(
+        this.format == 0,
+        Bytewise(ExtendedModelShortFormat[this["extended_items_count"]]),
+        Bytewise(ExtendedModelLongFormat[this["extended_items_count"]])),
+)
+
+CompositionDataPage1Element = Struct(
+    "number_s" / Rebuild(Int8ul, len_(this["sig_models"])),
+    "number_v" / Rebuild(Int8ul, len_(this["vendor_models"])),
+    "sig_models" / ModelRelationItem[this["number_s"]],
+    "vendor_models" / ModelRelationItem[this["number_v"]],
+)
+
+CompositionDataPage1 = Struct(
+    'element' / GreedyRange(CompositionDataPage1Element)
 )
 
 Retransmit = BitStruct(
-    # sssssccc
```

Comment:
This is not related to this PR, i prefer to keep it at is for better understanding of bitfield structure in this case